### ISSUE TITLE
We should change the defaults in bo_base

### ISSUE THREAD

**Issue by costashatz** [resibots/limbo#123]
Right now we have...

```cpp
// defaults
            struct defaults {
                typedef init::RandomSampling<Params> init_t; // 1

                typedef kernel::SquaredExpARD<Params> kf_t;
                typedef mean::Data<Params> mean_t;
                typedef model::GP<Params, kf_t, mean_t, model::gp::KernelLFOpt<Params>> model_t; // 2
                // WARNING: you have to specify the acquisition  function
                // if you use a custom model
                typedef acqui::GP_UCB<Params, model_t> acqui_t; // 3
                typedef boost::fusion::vector<stat::Samples<Params>, stat::AggregatedObservations<Params>, stat::ConsoleSummary<Params>> stat_t; // 4
                typedef boost::fusion::vector<stop::MaxIterations<Params>> stop_t; // 5
            };
```

I vote for:

- Keeping `RandomSampling` as initialization
- `kernel::Exp` for kernel
- Keeping `mean::Data`
- GP without optimization and put a default value of -1 in `boptimizer::hp_period` (already did it in [this](https://github.com/resibots/limbo/pull/120/files#diff-dbb952ae2ff475c414e3a669c041737cR19))
- `acqui::UCB` for acquisition function (just a preference - we could keep GP_UCB)

It's simpler and does not require too many things as defaults. I really do not like having kernel hyperparameters optimization as a default. What do you think?

**Comment by jbmouret** [comment]
OK for this. I agree with you.

**Comment by fedeallocati** [comment]
Just curious, why Exp kernel instead of Squared Exp?

2016-06-10 6:23 GMT-03:00 JB Mouret <notifications@github.com>:

> OK for this. I agree with you.
>
>
> On Fri, Jun 10, 2016 at 11:21 AM Konstantinos Chatzilygeroudis <

**Comment by costashatz** [comment]
> Just curious, why Exp kernel instead of Squared Exp?

Because basically if we use SquaredExpARD and we are not optimizing the hyperparameters, then it falls back to Exp with `l = 1` (if I am not mistaken - that's what I get when reading the code).

**Comment by costashatz** [comment]
Fixed in #138 ...
